component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java

```diff
@@ -201,6 +209,35 @@ protected void initiateAuthenticationRequest(HttpServletRequest request,
             if (context.isRetrying()) {
                 retryParam = "&authFailure=true&authFailureMsg=login.fail.message";
             }
+            IdentityErrorMsgContext errorContext = IdentityUtil.getIdentityErrorMsg();
+            IdentityUtil.clearIdentityErrorMsg();
+
+            String errorParam = StringUtils.EMPTY;
+            if (showAuthFailureReason) {
+                if (errorContext != null && errorContext.getErrorCode() != null) {
+                    log.debug("Identity error message context is not null.");
+                    String errorCode = errorContext.getErrorCode();
+                    if (errorCode != null) {
+                        String reason = null;
+                        if (errorCode.contains(":")) {
+                            String[] errorCodeWithReason = errorCode.split(":", 2);
+                            errorCode = errorCodeWithReason[0];
+                            if (errorCodeWithReason.length > 1) {
+                                reason = errorCodeWithReason[1];
+                            }
+                        }
+                        //Only adds error code if it is locked error code.
```

Comment:
`// Only adds error code if it is locked error code.`

Reply:
Addressed in https://github.com/wso2-extensions/identity-outbound-auth-totp/pull/156